app/num.py

```diff
@@ -14,10 +14,14 @@ def __init__(self, c: int = 0, s: str = ""):
         self.isSorted = True                        # no updates since last sort of data
         self.w = -1 if s.endswith("-") else 1       # check if ending with '-', return -1 if true, 1 otherwise
     
-     def nums(self):
-        if (!self.isSorted):
+    def nums(self):
+        if (not self.isSorted):
             self.has = sorted(self.has.items(), key=lambda x: x[1])     # sort the data first
             self.isSorted = True                                        # mark the isSorted flag true after sorting
         return self.has
 
-    
\ No newline at end of file
+    
+    def mid(self):
+
+        array = self.nums(self)
```

Comment:
array = self.nums() will also work
since you are already using self instance, no need to pass self again